CRM/CiviAwards/BAO/AwardDetail.php

```diff
@@ -76,4 +77,42 @@ private static function validateDates(array $params) {
     }
   }
 
+
+  /**
+   * Validates review_field options to ensure legal options are passed.
+   *
+   * @param array $params
+   *  Request Parameters
+   *
+   * @throws \Exception
+   */
+  private static function validateReviewFields(array &$params) {
+    $reviewFieldsLegalOptions = [
+      'id' => FILTER_VALIDATE_INT,
+      'weight' => FILTER_VALIDATE_INT,
+      'required' => FILTER_VALIDATE_BOOLEAN
+    ];
+    $reviewFields = CRM_Utils_Array::value('review_fields', $params);
+    if (is_array($reviewFields)) {
+      foreach ($reviewFields as &$reviewFieldOptions) {
+        $reviewFieldOptions = filter_var_array($reviewFieldOptions, $reviewFieldsLegalOptions, FALSE);
```

Comment:
Are we supposed to iterate over the `$reviewFieldOptions` at all? 
Looking at the manual here: https://www.php.net/manual/en/function.filter-var-array.php
The first array will contain the key => value pairs for the review fields but here the review fields are being iterated and the key is not even present.

Reply:
The thing is the `$reviewFields` come as `[['id' => 34, 'required' => true, 'weight' => 5], ['id' => 45, 'required' => false, 'weight' => 14]]`